Makefile

```diff
@@ -57,17 +57,73 @@ svsm.bin.elf: $(OBJS) src/start/svsm.lds
 %.lds: %.lds.S src/start/svsm.h
 	$(GCC) $(A_FLAGS) $(LDS_FLAGS) -E -P -o $@ $<
 
-prereq: .prereq
+prereq: prereq_rust
+	touch .prereq
 
-.prereq:
+prereq_rust:
 	curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
 	source $(HOME)/.cargo/env
 	echo "source $(HOME)/.cargo/env" >> ~/.bashrc
 	rustup component add rust-src
 	rustup component add llvm-tools-preview
 	cargo install xargo
 	cargo install bootimage
-	touch .prereq
+
+prereq_external:
+	git submodule update --init
+	(cd external/openssl && git checkout OpenSSL_1_1_1q  && \
+		./Configure \
+			--config=../openssl_svsm.conf \
+			SVSM \
+			no-afalgeng \
+			no-async \
+			no-autoerrinit \
+			no-autoload-config \
+			no-bf \
+			no-blake2 \
+			no-capieng \
+			no-cast \
+			no-chacha \
+			no-cms \
+			no-ct \
+			no-deprecated \
+			no-des \
+			no-dgram \
+			no-dsa \
+			no-dynamic-engine \
+			no-ec2m \
+			no-engine \
+			no-err \
+			no-filenames \
+			no-gost \
+			no-hw \
+			no-idea \
+			no-md4 \
+			no-mdc2 \
+			no-pic \
+			no-ocb \
+			no-poly1305 \
+			no-posix-io \
+			no-rc2 \
+			no-rc4 \
+			no-rfc3779 \
+			no-rmd160 \
+			no-scrypt \
+			no-seed \
+			no-sock \
+			no-srp \
+			no-ssl \
+			no-stdio \
+			no-threads \
+			no-ts \
+			no-whirlpool \
+			no-shared \
+			no-sse2 \
+			no-ui-console \
+			no-asm \
+			--with-rand-seed=none \
```

Comment:
The first patch in the series modifies the returned CPUID information to include the RDRAND and RDSEED bits, but then you don't use them here. You mention in the commit message that the system hangs, this is because the CPUID instruction will generate a #VC exception with error code 0x72, which the SVSM doesn't currently handle. You could add #VC handling for CPUID and allow the configuration to support RDRAND/RDSEED.

Reply:
@tlendacky thanks for reviewing the PR. That would certainly fix the cpuid hang or the inconsistency. On the other hand, we could just remove the RDRAND/RDSEED patch, it is not needed by the attestation report changes. What do you think?

The option `--with-rand-seed=rdcpu` calls the RDRAND/RDSEED backend to populate the entropy pool for DRBG, but we don't use the openssl DRBG.

IIRC, we used the RDRAND/RDSEED patch in the past only for our initial experiments with openssl, we don't need it anymore.